### PR TITLE
Sign In and Sign Up Feature

### PR DESCRIPTION
**Pull request by stecychirinda**
### Description
This pull request is implementing the core functionality for user authentication, including SignIn, SignUp, and LogOut features. The focus of this PR is on ensuring that both the backend logic and API endpoints are working correctly, as well as the frontend implementation.
In later PRs, we will be adding CSS styling to enhance the user interface and improve the overall user experience.

### Milestones
- Implementing user authentication (SignIn, SignUp, LogOut)
- Integrating with existing database schema
- Ensuring security best practices for password storage and validation
The work in this PR contributes towards the MVP requirement of user authentication, which is a critical component of our application.

### Resources
CodePath documentation
MDN Node.js documentation

### Test Plan
Testing was done using a combination of Insomnia to simulate API requests and manual testing through the frontend interface. Specifically:
- Tested successful SignIn and SignUp scenarios
- Verified that invalid credentials result in correct error responses
- Tested LogOut functionality to ensure session termination

Edge cases tested:
- Passwords ending with .edu (to test domain-specific password rules)
- Handling of duplicate usernames during SignUp
- Password length (at least 6 characters) validation during SignUp
- Session expiration and renewal
- Frontend testing included verifying that the UI updates correctly after successful login, logout, and error handling.

### CODE REVIEW

**Comment by jzysak25** [comment]
PR size can be reduced to two, even three preferably like:
Basic boiler template setup
Frontend implementation
Backend implementation
With each proper testing.
We can keep this PR as it is, but do use this comment at a reference for future PR size/scope

File: Backend/routes/Auth.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      !preferredContact ||
+      !profilePicture
+    ) {
+      return res.status(400).json({ message: "Missing required fields" });
+    }
+
+    if (password.length < 6) {
+      return res
+        .status(400)
+        .json({ message: "Password must be at least 6 characters" });
```
Comment (jzysak25) [inline Backend/routes/Auth.js:38]: `"Password must be at least ${minPWDLength} characters"`

File: Backend/routes/Auth.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    return res.status(401).json({ message: "Unauthorized" });
+  }
+  try {
+    const user = await prisma.user.findUnique({
+      where: { id: req.session.userId },
+      select: { email: true },
+    });
+    res.json({ id: req.session.userId, email: user.email });
+  } catch (err) {
+    console.log(err);
```
Comment (jzysak25) [inline Backend/routes/Auth.js:120]: Any reason you prefer console.log() over console.info()?

File: Frontend/my-vite-app/src/Components/CreateNewAccount.jsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+          value={password}
+          onChange={(e) => setPassword(e.target.value)}
+        />
+        <input
+          type="text"
+          placeholder="Major"
+          value={major}
+          onChange={(e) => setMajor(e.target.value)}
+        />
+        <div className="form-group">
```
Comment (jzysak25) [inline Frontend/my-vite-app/src/Components/CreateNewAccount.jsx]: Same thing, make the naming convention consistent
